src/DynamoCoreWpf/UI/Converters.cs

```diff
@@ -2341,4 +2349,292 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
+                // only for the case when expander is too far. (ex: 65,0,20,0)
+                if (left > right)
+                    right = left + 10;
+
+                // If both vertical and expander margins are not set (for root categories)                 
+                // then move the horizontal margin outside the outergrid. this is 
+                // used here, because we don't want the lines for root categories.                
+                if (left == 0 && expanderMargin.Right == 0)
+                    left = -10;
+
+                //if the vertical margin is not set, then move the horizontal line by
+                //10 points. This is mostly used for levels 0 or 1.
+                else if (left == 0)
+                {
+                    left = right + 10;
+                }
+
+                //If the treeview item is within 1 or 2 level, then move
+                //the horizontal line by 3 points. 
+                if (level >= 1 && level <= 2 && VerLnMargin.Left > 0)
+                {
+                    left = left - 3;
+                }
+
+                //for deep levels, use the margin same as vertical line
+                if (level > 2)
+                {
+                    left = VerLnMargin.Left;
+                }
+               
+                return new Thickness(left, 0, right, 0);               
+            }
+
+            public object[] ConvertBack(object value, Type[] targetTypes, object parameter, CultureInfo culture)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        /// This controls the Vertical line, when expanded / collapsed
+        /// </summary>
+        public class TreeViewVLineMarginConverter : IValueConverter
+        {
+            public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
+            {
+                Thickness margin = (Thickness)value;
+                int bottom = int.Parse(parameter.ToString());
+
+                //If the margin is not set
+                if (margin.Right == 0)
+                    return new Thickness(-10, 0, 0, 0);
+                
+                return new Thickness(margin.Left, margin.Top, margin.Right, bottom);
+                
+            }
+
+            public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        /// This controls the extra margin that is drawn even if the margin is not set
+        /// </summary>
+        public class TreeViewMarginCheck : IValueConverter
+        {
+            public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
+            {
+                Thickness margin = (Thickness)value;
```

Comment:
But Thickness is a non nullable value type. So, as cannot be used. But I will do a null check.